src/layers/sds/domain/changelog.py

```diff
@@ -50,13 +50,22 @@ def validate_change_number(
     def validate_changes(cls, changes):
         if isinstance(changes, bytes):
             changes = changes.decode("unicode_escape")
-        return changes
+        return changes.strip("\n")
 
     def changes_as_ldif(self) -> str:
-        change_lines = [
+        distinguished_name = dict(self.target_distinguished_name.parts)
+        unique_identifier_line = (
+            f"uniqueidentifier: {distinguished_name['uniqueidentifier']}"
+        )
+        header_lines = [
             f"dn: {self.target_distinguished_name.raw}",
             f"changetype: {self.change_type}",
         ]
-        if self.change_type is not ChangeType.DELETE:
-            change_lines.append(self.changes.strip("\n"))
+        if self.change_type is not ChangeType.ADD:
+            header_lines.append(f"{OBJECT_CLASS_FIELD_NAME}: {self.change_type}")
+
+        change_lines = header_lines + list(filter(bool, self.changes.split("\n")))
```

Comment:
Will this fail if changes is missing? I found one or two cases where instead of the changes object there looked to be an objectClass name (its int data so could be bogus but it happened at least once so I made changes optional above)

Reply:
the failure would happen in the parsing into a model, so for `SdsDeletionRequest` no changes are expected, `SdsModificationRequest` at least one field is expected and regular `add` expressions all the required fields are expected

but if any of those fail then it means that the ldif is bad, so we'd want it to fail at the model parsing at least